src-plugins/reformat/superResolutionProcess.cpp

```diff
@@ -75,81 +72,64 @@ QString superResolutionProcess::description(void) const
+        {
+            d->shapeBasedInterpolationFilter.setInput(res0);
+            d->shapeBasedInterpolationFilter.setInput(res1);
+            d->shapeBasedInterpolationFilter.setInput(res2);
+            d->shapeBasedInterpolationFilter.setInput(res3);
+
+            connect(&d->shapeBasedInterpolationFilter , SIGNAL(progressed(int)), this, SIGNAL(progressed(int)));
+            d->shapeBasedInterpolationFilter.run();
+
+            d->output = medAbstractDataFactory::instance()->create ("itkDataImageUChar3");
+            d->output->setData(d->shapeBasedInterpolationFilter.output()); // update output data
+            medUtilities::setDerivedMetaData(d->output, d->inputVol0, "superResolution");
+
+            return DTK_SUCCEED;
+        }
     }
-
-    d->d0 = castToUChar3(d->input0);
-    d->d1 = castToUChar3(d->input1);
-    d->d2 = castToUChar3(d->input2);
-    d->d3 = castToUChar3(d->input3);
-
-    if ( !d->d0 || !d->d1  || !d->d2  || !d->d3 )
-        return -1;
-
-    runButton<itk::Image<unsigned char, 3> >();
-
-    return EXIT_SUCCESS;
-}
-
-template <class ImageType> void superResolutionProcess::runButton()
-{
-    // compute process
-    typedef itk::Image<unsigned char, 3> MaskType;
-    typedef itk::CastImageFilter <ImageType, MaskType> CastFilter0;
-    typedef itk::CastImageFilter <ImageType, MaskType> CastFilter1;
-    typedef itk::CastImageFilter <ImageType, MaskType> CastFilter2;
-    typedef itk::CastImageFilter <ImageType, MaskType> CastFilter3;
-
-    typename CastFilter0::Pointer castFilter0 = CastFilter0::New();
-    castFilter0->SetInput(0,dynamic_cast<ImageType*>((itk::Object*)(d->d0->data())));
-    castFilter0->Update();
-
-    typename CastFilter1::Pointer castFilter1 = CastFilter1::New();
-    castFilter1->SetInput(0,dynamic_cast<ImageType*>((itk::Object*)(d->d1->data())));
-    castFilter1->Update();
-
-    typename CastFilter2::Pointer castFilter2 = CastFilter2::New();
-    castFilter2->SetInput(0,dynamic_cast<ImageType*>((itk::Object*)(d->d2->data())));
-    castFilter2->Update();
-
-    typename CastFilter3::Pointer castFilter3 = CastFilter3::New();
-    castFilter3->SetInput(0,dynamic_cast<ImageType*>((itk::Object*)(d->d3->data())));
-    castFilter3->Update();
-
-    d->output = medAbstractDataFactory::instance()->create ("itkDataImageUChar3");
-
-    shapeBasedInterpolation shapeBasedInterpolationFilter;
-
-    shapeBasedInterpolationFilter.setInput(castFilter0->GetOutput(), castFilter1->GetOutput(), castFilter2->GetOutput(), castFilter3->GetOutput());
-    connect(&shapeBasedInterpolationFilter , SIGNAL(progressed(int)), this, SIGNAL(progressed(int)));
-    shapeBasedInterpolationFilter.run();
-    d->output->setData(shapeBasedInterpolationFilter.output()); // update output data
+    return DTK_FAILURE;
 }
 
-medAbstractData* superResolutionProcess::castToUChar3(medAbstractData* image)
+MaskType* superResolutionProcess::castToUChar3(dtkSmartPointer<medAbstractData> image)
```

Comment:
If we could remove the word cast, yes, it's a good idea since it is technically not a cast, but a cast of a member object.
As I said, the ```cast``` template function should also be renamed (you can give it the same name since it does the same thing, only for one type).